### PR TITLE
Allow suffix-specific output directories for a project

### PR DESCRIPTION
**Pull request by jyhsu**
Hello @nicks,

Please review the following commits I made in branch 'jean-version'.

ec84867d5df42e525abaf80f05db4077bd552011 (2015-07-06 15:47:10 -0700)
Allow suffix-specific output directories for a project

R=@nicks

### CODE REVIEW

File: tests/project_test.js [hunk 1]
```diff
@@ -69,6 +69,43 @@ exports.testBasicCompilation = function (test) {
...
+      test.equals(path.join(__dirname, 'java/generated-stuff', 'common.proto.java'), compilations[1].fileName)
+      test.equals(path.join(__dirname, 'java/generated-stuff', 'person.proto.java'), compilations[2].fileName)
+      test.equals(path.join(__dirname, 'generated-stuff', 'vehicle.proto.xx.js'), compilations[3].fileName)
+      test.equals(path.join(__dirname, 'generated-stuff', 'common.proto.xx.js'), compilations[4].fileName)
+      test.equals(path.join(__dirname, 'generated-stuff', 'person.proto.xx.js'), compilations[5].fileName)
+
+      test.done()
+    })
+    .fail(function (err) {
+      console.log(err.stack)
```
Comment (nicks) [inline tests/project_test.js]: the standard way to do this is
```
test.ifError(err)
test.done()
```

otherwise the test framework will just hang

Comment (nicks) [inline tests/project_test.js]: (we wrote nodeunitq to make this automatic)

**Comment by nicks** [comment]
lgtm

File: tests/project_test.js (hunk 1 above)
Comment (jyhsu) [inline tests/project_test.js]: cool converted the rest of the tests!

**Pull request merged**